synapse/handlers/pagination.py

```diff
@@ -131,7 +131,7 @@ def purge_history_for_rooms_in_range(self, min_ms, max_ms):
             min_ms, max_ms, include_null
         )
 
-        for room_id, retention_policy in rooms:
+        for room_id, retention_policy in rooms.items():
```

Comment:
you probably want `six.iteritems(rooms)` (or whatever it's called) for python2 reasons